src/engine/controls/cuecontrol.cpp

```diff
@@ -594,13 +616,18 @@ void CueControl::hotcueSet(HotcueControl* pControl, double v) {
     pCue->setHotCue(hotcue);
     pCue->setLabel("");
     pCue->setType(mixxx::CueType::HotCue);
-    // TODO(XXX) deal with spurious signals
-    attachCue(pCue, pControl);
 
-    if (getConfig()->getValue(ConfigKey("[Controls]", "auto_hotcue_colors"), false)) {
-        const QList<PredefinedColorPointer> predefinedColors = Color::kPredefinedColorsSet.allColors;
-        pCue->setColor(predefinedColors.at((hotcue % (predefinedColors.count() - 1)) + 1));
+    auto hotcueColorPalette = m_colorPaletteSettings.getHotcueColorPalette();
+    ConfigKey autoHotcueColorsKey("[Controls]", "auto_hotcue_colors");
+    int iColorIndex = 0;
```

Comment:
Here we need to apply the default white or whatever the user has decided to choose. 

Unfortunately we cannot postpone this to another PR because it will mass up users Database if red is introduced here. 
Alterntive we may create a branch in mixxx/mixxxdj to collect the changes.

Reply:
Why does it mess up the Database? This only applies a default color to new cues.

I admit it's a bit inconvenient for people that want to use a different, fixed color because they need to edit them manually or wait until the mass replace tool has been merged, but IMHO inconvenience is expected when you use master.

I split the PRs to keep stuff reviewable. I can also implement the preferences stuff on this branch, but that'll probably hundreds or thousands of additional lines that could also go into a separate PR IMHO.

Reply:
Using here the default color is not a big change.

For the next step, we may re-introduce the combo box for selecting the default color.

You find most if the code in my branch.

I would expect that all cues, that are on the default color will change as well.
But I think there are different opinions.

So we may poop up ad dialog asking for that. 

Once this is introduced we can merge it.

Reply:
By the way, you next branch on top of this is already hard to review, because all changes from here are also shown in the diff. If you like to keep them separate merging them into a mixxxdj branch would be helpful.

Reply:
I'm still not convinced that we need to add a way to edit the default color as part of this PR.

Before merging this PR:
  - Colors for new cues are set to "no color" (displayed color is skin dependent) or by index if `auto_hotcue_colors` is enabled

Before merging this PR:
  - Colors for new cues are set to "red" or by index if `auto_hotcue_colors` is enabled

I don't really see how this is fundamentally different. Users that care about the color need to edit the color after setting a cue anyway (or have `auto_hotcue_colors` enabled). And If the user was using the LateNight skin, there won't even be a noticeable difference because "No color" was mapped to "red" anyway.

> Using here the default color is not a big change.
> For the next step, we may re-introduce the combo box for selecting the default color.
> You find most if the code in my branch.
> I would expect that all cues, that are on the default color will change as well.
> [...]
> So we may poop up ad dialog asking for that.
> Once this is introduced we can merge it.

That is a quite a bit of additional work that would be removed in the next PR anyway. I'm not keen on adding that just as a convenience for master users, especially since the current implementation only affects new Cues and IMHO doesn't have a negative impact on usability.

> By the way, you next branch on top of this is already hard to review, because all changes from here are also shown in the diff. If you like to keep them separate merging them into a mixxxdj branch would be helpful.

Doesn't really make sense to review this right now, it's still WIP and I might rebase it.

Reply:
There a many users not caring about the colors. After this branch is merged as it is. These user has two different colores in his DB, white and red. 
This state makes it difficult to change later to one single color.

We have also the issue of alpha users.
This change introduces two default colors probably clashing with meaningful colors if the users.

All this discussion is pointless, if we merge to a new mixxxdj branch which is done in seconds.

Regarding the additional work, this would allow use to release 2.3.
So it might be sensible to do.
All the palette things can be postponed to 2.4.

Reply:
> There a many users not caring about the colors. After this branch is merged as it is. These user has two different colores in his DB, white and red.
> This state makes it difficult to change later to one single color.

Ah okay, now I understand the problem. This would still be easily fixable with the mass replacement tool, but we should change this anyway.

Let's agree on a single, fixed "no color" replacement on Zulip that we'll just for the schema migration, but also for newly set cues. That would mitigate the problem without too much work.

> Regarding the additional work, this would allow use to release 2.3.
So it might be sensible to do.
All the palette things can be postponed to 2.4.

I looked into the GUI parts of Qt in the last few days, played around with some palette editor designs and implemented some basic editor design. It was actually less work than I anticipated. I can push it for review, maybe we don't even need to postpone.